main.js

```diff
@@ -1,29 +1,136 @@
-// Button Variables
-const homeButton = document.querySelector('#btn-home')
-const bookButton = document.querySelector('#btn-bookmark')
-const createButton = document.querySelector('#btn-create')
+const homeButton = getEl('#btn-home')
+const bookButton = getEl('#btn-bookmark')
+const createButton = getEl('#btn-create')
 
-// Page Variables
-const indexPage = document.querySelector('#questions')
-const bookmarkPage = document.querySelector('#bookmarks')
-const createPage = document.querySelector('#create')
+const indexPage = getEl('#questions')
+const bookmarkPage = getEl('#bookmarks')
+const createPage = getEl('#create')
 
-// Functions
+function getEl(selector) {
+  const el = document.querySelector(selector)
+  return el
+}
+
+function hideAll() {
+  const mainElArray = document.querySelectorAll('.main')
+  mainElArray.forEach(mainEl => {
+    mainEl.classList.add('main--hidden')
+  })
+}
 
 homeButton.addEventListener('click', () => {
+  hideAll()
   indexPage.classList.remove('main--hidden')
-  bookmarkPage.classList.add('main--hidden')
-  createPage.classList.add('main--hidden')
 })
 
 bookButton.addEventListener('click', () => {
-  indexPage.classList.add('main--hidden')
+  hideAll()
   bookmarkPage.classList.remove('main--hidden')
-  createPage.classList.add('main--hidden')
 })
 
 createButton.addEventListener('click', () => {
-  indexPage.classList.add('main--hidden')
-  bookmarkPage.classList.add('main--hidden')
+  hideAll()
   createPage.classList.remove('main--hidden')
 })
+
+// show and hide answer
+
+const cardButton = document.querySelectorAll('.card__button')
+cardButton.forEach(function (show) {
+  show.addEventListener('click', () => {
+    show.nextElementSibling.classList.toggle('card__answer--hidden')
+  })
+})
+
+const allbookElements = document.querySelectorAll('.card__bookmark')
+
+allbookElements.forEach(function (si) {
+  si.addEventListener('click', () => {
+    si.classList.toggle('card__bookmark--unchecked')
+  })
+})
+
+// clear all input-Elements in a Form
+
+const buttonClear = document.querySelector('.btn__submit')
+
+buttonClear.addEventListener('click', () => {
+  document.querySelector('.form').reset()
+})
```

Comment:
Please have a look at this codepen and my solution: https://codepen.io/macfrei/pen/mdwOEBd?editors=0010

I know we haven't discussed that in detail in the course 🙂